metals/src/main/resources/MetalsSbtBsp.scala

```diff
@@ -0,0 +1,64 @@
+package metals
+
+import sbt._
+import sbt.Keys._
+import Project.inConfig
+import sbt.internal.inc.ScalaInstance
+
+object MetalsPlugin extends AutoPlugin {
+  override def requires = plugins.JvmPlugin
+  override def trigger = allRequirements
+
+  val semanticdbVersion = "4.3.21"
+
+  override lazy val projectSettings: Seq[Def.Setting[_]] = Seq(
+    semanticdbCompilerPlugin := {
+      ("org.scalameta" % "semanticdb-scalac" % semanticdbVersion)
+        .cross(CrossVersion.full)
+    },
+    allDependencies ++= {
+      val sv = scalaVersion.value
+      if (!ScalaInstance.isDotty(sv)) {
+        List(
+          compilerPlugin(
+            "org.scalameta" % s"semanticdb-scalac_${sv}" % semanticdbVersion
+          )
+        )
+      } else Nil
+    }
+  ) ++ inConfig(Compile)(configurationSettings) ++ inConfig(Test)(
+    configurationSettings
+  )
+
+  lazy val configurationSettings: Seq[Def.Setting[_]] = List(
+    scalacOptions := {
+      val old = scalacOptions.value
+      val sdbOptions = semanticdbOptions.value
+      val sv = scalaVersion.value
+      (old.toVector ++ sdbOptions ++
+        (if (ScalaInstance.isDotty(sv)) Some("-Ysemanticdb")
+         else None)).distinct
+    },
+    semanticdbTargetRoot := {
+      val in = semanticdbIncludeInJar.value
+      if (in) classDirectory.value
+      else semanticdbTargetRoot.value
+    },
+    semanticdbOptions ++= {
```

Comment:
Do you know what is the purpose of having the separate `semanticdbOptions` ?

Reply:
I see it's coming from sbt, but I am honestly just wondering.

Reply:
To be honest, not really. Some of this was more or less of a shameless copy pasta from here https://github.com/sbt/sbt/blob/85a0d08a4eb424f88332bcb1f80c1a69344f1e30/main/src/main/scala/sbt/plugins/SemanticdbPlugin.scala.

I'll play around with this a bit more, see what is included already, and see what the diff is vs me including them here.

Reply:
I think you shouldn't bother. If it does work, it should be perfectly fine to leave as is. That setting does exist, so most likely we should use it. I was just curious.